src/tags/bool.ts

```diff
@@ -0,0 +1,129 @@
+import { Cluster } from '../cluster';
+import { BaseSubtag, BBTagContext, SubtagCall } from '../core/bbtag';
+import { SubtagType, parse, bbtagUtil, compare } from '../utils';
+
+const operators = [
+    '==',
+    '!=',
+    '>=',
+    '>',
+    '<=',
+    'startswith',
+    'endswith',
+    'includes',
+    'contains'
+];
+
+interface OperatorHandler {
+    [index: string]: (a: string, b: string) => boolean;
+}
+
+export class BoolSubtag extends BaseSubtag {
+    public operatorHandler: OperatorHandler;
+    public operators: string[];
+    public constructor(cluster: Cluster) {
+        super(cluster, {
+            name: 'bool',
+            category: SubtagType.COMPLEX,
+            definition: [
+                {
+                    args: ['arg1', 'evaluator', 'arg2'],
+                    description:
+                        'Evaluates `arg1` and `arg2` using the `evaluator` and returns `true` or `false`. ' +
+                        'Valid evaluators are `' +
+                        operators.join('`, `') +
+                        '`\n' +
+                        'The positions of `evaluator` and `arg1` can be swapped.',
+                    exampleCode: '{bool;5;<=;10}',
+                    exampleOut: 'true',
+                    execute: (ctx, args, subtag) =>
+                        this.runCondition(
+                            ctx,
+                            subtag,
+                            args[0].value,
+                            args[1].value,
+                            args[2].value
+                        )
+                }
+            ]
+        });
+        //TODO possibly remove this?
+        this.operators = operators;
+        /**
+         * * There has to be a way to better define these types??
```

Comment:
I agree, these dont need to be here.  One idea cat and I were throwing around before was an operator subtag
`{>;x;y}` something like this which can be referenced by `{bool}`, `{logic}`, `{if}`, `{math}` etc. Maybe this would be a good time to implement this?
Move the operator logic to util, have the operators grouped by category (bool, numeric, compare).

`{bool}` and `{if}` would use the `compare` operators (`==`, `!=`, `<`, `>` etc.)
`{logic}` would use the `bool` operators (`||`, `&&`, `!` etc)
`{math}` would use the `numeric` operators (`+`, `-`, `*` etc)
`{operator}` would use all the operators (`{+;5;6;7;8}`, `{*;10;9;8;7}`, `{!;true}`, `{>;a;b}` etc

Reply:
Sounds like a really good idea, I will be sure to update this PR with it when I have the time.

For future commands/subtags, should 'shared' code just always be moved to utils so you wouldn't have stuff like `this.subtags.get('bool').methodHere()` or like `this.commands.get('ban').ban()`?

Reply:
I think if its shared code and doesnt rely on anything else (discord, database etc) then yes, move to util following the pattern there. otherwise, if its only used on the clusters move it to ClusterUtilities, if its used by the master process and the clusters then move it to BaseUtilities